schemas/littlehorse/service.proto

```diff
@@ -822,36 +822,21 @@ message PrincipalIdList {
   optional bytes bookmark = 2;
 }
 
-// EVOLVING: Search for ExternalEvent's.
-//
-// Currently, this request allows you to search for ExternalEvent's based on either:
-// 1. A WfRunId
-// 2. An ExternalEventDefName and status.
-//
-// This specific RPC is under discussions for a Refactor and will soon experience breaking changes.
-// It is recommended for applications needing to search by WfRunId to instead use the
-// `rpc ListExternalEvents` call, as we plan to remove the 'by WfRunId' option from this request.
-//
-// Specifically, we plan to remove the "by wfRunId" option (which is redundant with the ListExternalEvents
-// request), and "flatten" the "ByExtEvtDefNameAndStatusRequest" fields into the main message.
+// Search for ExternalEvents based on certain criteria.
 message SearchExternalEventRequest {
   // Bookmark for cursor-based pagination; pass if applicable.
   optional bytes bookmark = 1;
   // Maximum results to return in one request.
   optional int32 limit = 2;
 
-  // EVOLVING: message encapsulating criteria to search for ExternalEvent's by their status
-  // and ExternalEventDefName.
-  message ByExtEvtDefNameAndStatusRequest {
-    string external_event_def_name = 1;
-    optional bool is_claimed = 2;
-  }
+  // Specifies to return only Principals's created after this time
+  optional google.protobuf.Timestamp earliest_start = 3;
 
-  // EVOLVING
-  oneof ext_evt_criteria {
-    WfRunId wf_run_id = 3;
-    ByExtEvtDefNameAndStatusRequest external_event_def_name_and_status = 4;
-  }
+  // Specifies to return only Principals's created before this time
+  optional google.protobuf.Timestamp latest_start = 4;
+
+  // Search for external event definitions by Id
```

Comment:
We're not searching for external event definitions. We're searching for `ExternalEvent`s.
> This field denotes the `ExternalEventDef` of the `ExternalEvent`s that we are searching for. This field is mandatory.